src/common/src/cache.rs

```diff
@@ -769,23 +770,45 @@ impl<K: LruKey, T: LruValue> LruCache<K, T> {
 
 pub struct CleanCacheGuard<'a, K: LruKey + Clone + 'static, T: LruValue + 'static> {
     cache: &'a Arc<LruCache<K, T>>,
-    key: K,
+    key: Option<K>,
     hash: u64,
-    success: bool,
+}
+
+impl<'a, K: LruKey + Clone + 'static, T: LruValue + 'static> CleanCacheGuard<'a, K, T> {
+    fn mark_success(&mut self) -> K {
```

Comment:
A more rust way is to take the ownership of `self`, because `mark_success` means the guard has finished its work.

Reply:
> A more rust way is to take the ownership of `self`, because `mark_success` means the guard has finished its work.

fixed